docs/release-notes/charmcraft-3.5.rst

```diff
@@ -0,0 +1,161 @@
+Charmcraft 3.5 release notes
+============================
+
+**UNRELEASED**. Expected release date: 7 April 2025
+
+Learn about the new features, changes, and fixes introduced in Charmcraft 3.5.
+
+
+Requirements and compatibility
+------------------------------
+
+For development and testing, Charmcraft requires a host with a minimum of 4GB RAM
+running a Linux distribution compatible with systemd.
+
+All versions of Charmcraft require the following software:
+
+- systemd
+- `snapd`_
+- Either `LXD`_ or `Multipass`_
+
+We recommend you install the `Charmcraft snap <https://snapcraft.io/charmcraft>`_. It
+comes comes bundled with all its dependencies.
+
+Non-snap installations of Charmcraft have the following dependencies:
+
+- Python 3.10 or higher
+- libgit2 1.7
+- `skopeo`_
+- `Spread`_
+
+
+What's new
+----------
+
+Charmcraft 3.5 brings the following features, integrations, and improvements.
+
+
+Updated ``init`` templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The templates generated by ``charmcraft init`` now use the new
+:doc:`/reference/platforms` format.
+
+
+JSON Schema
+~~~~~~~~~~~
+
+The `JSON Schema Store <https://schemastore.org>`_ now includes a schema for the :ref:`charmcraft-yaml-file`. If your text editor supports getting JSON schemas from
+the schema store, they may already be visible in your text editor.
+
+.. image:: /_static/assets/json-schema-example-light.webp
+   :width: 451
+   :alt: A charmcraft.yaml file open in kate editor showing the mouseover hint for the parts key.
+
+If your editor doesn't support JSON schemas directly but does support the
+`Language Server Protocol <https://microsoft.github.io/language-server-protocol/>`_,
+you can use a `YAML Language Server <https://snapcraft.io/yaml-language-server>`_
+to provide the feature.
```

Comment:
> doesn't support JSON schemas directly

I'm not sure what this means, or whether it's technically correct. Any tool with autocomplete has the rough notion of a language server (a separate module that provides intelligent language parsing), even if they don't use that specific modelling.

Ultimately I don't think it's a distinction worth making in this note. I think we're pushing too far from the topic at hand.

If it's a question of "does my IDE support JSON schemas?", let's leave it to the reader, as you did in the first paragraph. The title of the note will signal to them a topic they may or may not understand, with a feature they may or may not use.